tests/check_framework/test_model_checks.py

```diff
@@ -50,6 +50,25 @@ class Meta:
             )
         ])
 
+    def test_collision_across_apps_special(self):
+        constraint = models.CheckConstraint(check=models.Q(id__gt=0), name='%(app_label)s_%(class)s_foo')
+        index = models.Index(fields=['id'], name='%(app_label)s_%(class)s_index')
+
+        class AbstractModel(models.Model):
+            class Meta:
+                app_label = 'basic'
```

Comment:
I added it to get rid of the "names cannot be longer than 30 characters" error.

Reply:
Which proves that it doesn't work properly because names of indexes should be
`check_framework_model1_index` and `check_framework_model2_index`, currently it is `check_framework_abstractmodel_index` in both cases.

Reply:
fixed for abstract methods and added new tests.